lib/identity_cache/query_api.rb

```diff
@@ -476,5 +484,15 @@ def was_new_record? # :nodoc:
       pk = self.class.primary_key
       !destroyed? && transaction_changed_attributes.has_key?(pk) && transaction_changed_attributes[pk].nil?
     end
+
+    def load_and_readonlyify(association_name)
+      record_or_records = send(association_name)
+      if self.class.reflect_on_association(association_name).collection?
+        record_or_records.map(&:readonly!)
+      else
+        record_or_records.try(:readonly!)
```

Comment:
This is marking records as read-only in the active record association, which could end up having a surprising and annoying side-effect when the association is ever accessed without using IDC.  Accessing the active record association directly shouldn't ever be returning a record from the cache, so it would be safe to modify and save it, but this would break code that does that.

E.g. this doesn't seem like a desirable behaviour

```ruby
item = Item.find(item_id)
uncached_record = item.associated_record
assert_equal false, uncached_record.readonly?
item.fetch_associated_record
assert_equal true, uncached_record.readonly?
```